File: backend/src/main/java/com/example/store_manager/model/Payment.java
package com.example.store_manager.model;

import java.math.BigDecimal;
import java.time.Instant;

import org.hibernate.envers.Audited;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Version;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Audited
@EntityListeners(AuditingEntityListener.class)
@Entity
@Table(name = "payments")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(optional = false)
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;

    // Stripe or provider reference
    @Column(name = "provider_payment_id")
    private String providerPaymentId;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal amountTotal;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal platformFee;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal shopAmount;

    @Column(nullable = false)
    private String currency;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentStatus status;

    // Link to payout batch (nullable until paid out)
    @ManyToOne
    @JoinColumn(name = "payout_id")
    private Payout payout;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @Version
    private Long version;

}

File: frontend/src/types/payment.ts
export type PaymentStatus =
  | "PENDING"
  | "SUCCEEDED"
  | "FAILED"
  | "REFUNDED"
  | "PARTIALLY_REFUNDED";

export interface PaymentResponseDto {
  id: number;

  orderId: number;

  providerPaymentId?: string;

  amountTotal: number;
  platformFee: number;
  shopAmount: number;

  currency: string;

  status: PaymentStatus;

  createdAt: string;
}

export interface PaymentDto {
  id: number;
  orderId: number;
  amountTotal: number;
  platformFee: number;
  shopAmount: number;
  currency: string;
  status: PaymentStatus;
  createdAt: string;
}

export interface PaymentListResponseDto {
  payments: PaymentResponseDto[];
  total: number;
}
